Add Create Activity popup

jobron opened the pull request:
Closes #24

jobron committed:
Start add new activity popup

jobron committed:
Merge branch 'main' into SportsPageChanges

jobron committed:
Merge branch 'main' into SportsPageChanges

jobron committed:
Make popup style consistent, add remaining fields

jobron committed:
Initial activity POST with popup fixes

Drayz64 commented on spector/frontend/src/css/addActivityPopup.css:
```diff
@@ -0,0 +1,92 @@
+}
+  
+  /* X close button */
+.modal > .close {
```
The CSS files are global. So as long as a CSS file is imported in one JS file, you can use its styles in any of the other JS files, without needing to import the CSS file.

So if these popup styles will be the same as in activityPopup.css then we don't need them to be duplicated. 

But if they differ then
- give them a unique name like `.Create-modal`
- or use a css module so the styles are local
```js
// Using a CSS module
import styles from "css/addActivityPopup.module.css";

<button className={styles.buttonAdd}>
    Add a New Event...
</button>
```

jobron committed:
Switch to duration for activity popup

jobron committed:
Merge branch 'main' into SportsPageChanges

jobron committed:
Fix styling and pass sport to popup

this is definitely not correct :+1:

jobron committed:
Reload page after submitting (oops)

sorry

Commit:
A few changes:

- sportsName is now passed to addActivityPopup.js without using the spread operator
- Set a minimum date (of today) for the dateTime picker
- Prettier also did some automatic prettying

Commit:
Added an exmaple forms.css

Drayz64 commented:
There's a styling issue with the sign in and register pages:

![image](https://user-images.githubusercontent.com/65258846/159353471-29df5b35-a91c-4e84-9013-f627aa273bbd.png)

I think this happens because `input[type="text"]` inside [addActivityPopup.css](https://github.com/Valentin349/Sport-Societies-Network/blob/SportsPageChanges/spector/frontend/src/css/addActivityPopup.css#L59) overrides/merges with `input[type="text"]` inside [signUp.css](https://github.com/Valentin349/Sport-Societies-Network/blob/SportsPageChanges/spector/frontend/src/css/components/signUp.css#L26) (as well as [signIn.css](https://github.com/Valentin349/Sport-Societies-Network/blob/SportsPageChanges/spector/frontend/src/css/components/signIn.css#L26))

- If the styles for `labels` and the 3 different `inputs` can be the same for sign in/register and addActivityPopup then put them in [forms.css](https://github.com/Valentin349/Sport-Societies-Network/blob/SportsPageChanges/spector/frontend/src/css/components/forms.css) and import it inside of index.js

- If not, then instead of giving all labels the same style, can you apply a `className` to each label and style them that way?

jobron committed:
Styling improvements

Move to forms, use explicit centering in div instead of margins in signIn/signUp

jobron commented:
![image](https://user-images.githubusercontent.com/80117666/159742069-e7937b95-2056-4404-85a9-a0717aa19014.png)
![image](https://user-images.githubusercontent.com/80117666/159742190-ed45e375-5068-49a2-b62d-a446537c5197.png)

There's still some problems with resizing when windows get small but this should make the pull request servicable. Everything is now in forms.css too. Might need some prettying.

The problem wasn't too much styling but the activities and sign in pages use different ways to centre the inputs, sign in uses margin but activities uses an explicit div with centering styling. Now they both use the div instead. I put both in forms but they're not as global as the other styles in there so lmk if you want it changed.

Only issue remaining is that the text in the sign in page is black instead of white, but I'm not sure the easiest way to correct that outside of giving everything classnames and disregarding the label class altogether.
